### PR TITLE
Add support in websocket sub-protocols

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

**Comment by samhed** [comment]
Just curious - what are you using this for?

**Comment by shiramax** [comment]
@samhed I need to open a vnc console for kubevirt VM, In order to do that, I have to authenticate with kubevirt-api and therefore I have to send bearer token.
there are two options to send this bearer token,
One way is via a header, However that will not work from browsers since they can't add headers when upgrading. The second option is to use a special subprotocol format to transfer the bearer token.

File: docs/EMBEDDING.md [hunk 1]
```diff
@@ -67,6 +67,8 @@ query string. Currently the following options are available:
 * `logging` - The console log level. Can be one of `error`, `warn`, `info` or
   `debug`.
 
+* `wsProtocols` - websocket sub-protocols
```
Comment (samhed) [inline docs/EMBEDDING.md]: I would like us to use proper capitalization, write 'WebSocket' please.

File: docs/EMBEDDING.md [hunk 2]
```diff
@@ -67,6 +67,8 @@ query string. Currently the following options are available:
 * `logging` - The console log level. Can be one of `error`, `warn`, `info` or
   `debug`.
 
+* `wsProtocols` - WebSocket sub-protocols
```
Comment (CendioOssman) [inline docs/EMBEDDING.md]: `EMBEDDING.md` is for `vnc.html`, which I can't see any changes for? So either omit this part, or make sure `vnc.html` can pass it on to the `RFB` constructor.

File: core/rfb.js [hunk 3]
```diff
@@ -53,6 +53,7 @@ export default class RFB extends EventTargetMixin {
         this._shared = 'shared' in options ? !!options.shared : true;
         this._repeaterID = options.repeaterID || '';
         this._showDotCursor = options.showDotCursor || false;
+        this._wsProtocols = options.wsProtocols || ['binary'];
```
Comment (CendioOssman) [inline core/rfb.js]: Argh. Are we still including that `"binary"` default. That is very non-standard.

@samhed, could you remove that and just have nothing as the default once this is merged?

Comment (samhed) [reply core/rfb.js]: Sure

**Comment by shiramax** [comment]
@CendioOssman @samhed thanks, could you please review again?

File: core/rfb.js (hunk 3 above)
Comment (samhed) [reply core/rfb.js]: Fixed in e14aa4d0fe43a939378a3dd7666806fa7018dfe7

**Pull request merged**